test/python/dagcircuit/test_dagcircuit.py

```diff
@@ -248,6 +248,140 @@ def test_addding_individual_bit(self):
+        self.dag.remove_cregs(self.creg0)
+
+        # creg removal always succeeds
+        self.assert_cregs_is(self.original_cregs, excluding={self.creg0})
+
+        # creg0 is gone, leaving both creg0[0] and creg0[1] unreferenced.
+        # however, only creg0[1] is idle, and thus removed.
+        self.assert_clbits_is(self.original_clbits)
+
+    def test_remove_cregs_shared_bits(self):
+        """Removing a classical register does not remove other classical
+        registers that are simply aliases of it."""
+        self.dag.remove_cregs(self.creg1)
+
+        # Only creg1 should be deleted, not its alias
+        self.assert_cregs_is(self.original_cregs, excluding={self.creg1})
+        self.assert_clbits_is(self.original_clbits)
+
+    def test_remove_unknown_creg(self):
+        """Classical register removal of unknown registers raises."""
+        unknown_creg = ClassicalRegister(1)
+        with self.assertRaises(DAGCircuitError):
+            self.dag.remove_cregs(unknown_creg)
+
+        self.assert_cregs_is(self.original_cregs)
+        self.assert_clbits_is(self.original_clbits)
+
+    def test_remove_idle_clbit(self):
+        """Idle classical bits not referenced by any register can be removed."""
+        self.dag.remove_clbits(self.individual_clbit)
+
+        self.assert_cregs_is(self.original_cregs)
+        self.assert_clbits_is(self.original_clbits, excluding={self.individual_clbit})
+
+    def test_remove_busy_clbit(self):
+        """Classical bit removal of busy classical bits raises."""
+        self.dag.apply_operation_back(Measure(), [self.qreg[0]], [self.individual_clbit])
+
+        with self.assertRaises(DAGCircuitError):
+            self.dag.remove_clbits(self.individual_clbit)
+
+        self.assert_cregs_is(self.original_cregs)
+        self.assert_clbits_is(self.original_clbits)
+
+    def test_remove_referenced_clbit(self):
+        """Classical bit removal removes registers that reference a removed bit,
+        even if they have other bits that aren't removed."""
+        self.dag.remove_clbits(self.creg0[0])
+
+        self.assert_cregs_is(self.original_cregs, excluding={self.creg0})
+        self.assert_clbits_is(self.original_clbits, excluding={self.creg0[0]})
+
+    def test_remove_multi_reg_referenced_clbit(self):
+        """Classical bit removal removes all registers that reference a removed bit."""
+        self.dag.remove_clbits(*self.creg1)
+
+        self.assert_cregs_is(self.original_cregs, excluding={self.creg1, self.creg1_alias})
+        self.assert_clbits_is(self.original_clbits, excluding=set(self.creg1))
+
+    def test_remove_unknown_clbit(self):
+        """Classical bit removal of unknown bits raises."""
+        unknown_clbit = Clbit()
+        with self.assertRaises(DAGCircuitError):
+            self.dag.remove_clbits(unknown_clbit)
+
+        self.assert_cregs_is(self.original_cregs)
+        self.assert_clbits_is(self.original_clbits)
+
```

Comment:
Do you want to test removal of an unknown register as well?

Reply:
This is done on line 334 :).

Reply:
Oh, so it is.  I must have forgotten!